python/tests/test_pyvw.py

```diff
@@ -122,13 +123,23 @@ def test_multilabel_prediction_type():
 
 def test_cbandits_label():
     model = vw(cb=4, quiet=True)
-    assert pyvw.cbandits_label(model.example('1 |')).costs[0].label == 1
+    cbl = pyvw.cbandits_label(model.example('1 |'))
+    assert cbl.costs[0].label == 1
+    assert cbl.costs[0].probability == 0.0
+    assert cbl.costs[0].partial_prediction == 0.0
+    assert cbl.prediction == 0
+    assert str(cbl) == '1:3.4028234663852886e+38'
```

Comment:
Why does the produced string contain `3.4028234663852886e+38`? I don't see this value in the above assertions

Reply:
Let's get all tests in CI to pass, for example we're getting failures here https://dev.azure.com/vowpalwabbit/Vowpal%20Wabbit/_build/results?buildId=5684&view=logs&j=57e19422-7def-5009-e470-a832cad5a7e3&t=8e4bc9ff-39a8-50a5-9cb9-43a19053be7e

Let us know if you need help figuring out the test results.

Reply:
The Test failure log shows this `AssertionError: assert '1:3.40282346639e+38' == '1:3.4028234663852886e+38'`. Maybe this can be because of some round-off, also I use Python3 whereas I saw that the tests run on Python 2. 
Locally all the tests pass on my computer.

Reply:
Locally when I changed according to the output of log, I get:
```
>       assert str(csl) == '1:3.40282346639e+38'
E       AssertionError: assert '1:3.4028234663852886e+38' == '1:3.40282346639e+38'
E         - 1:3.4028234663852886e+38
E         ?               ^^^^^^
E         + 1:3.40282346639e+38
E         ?               ^

```

Reply:
Comparing floats by string is error prone due to precision differences between environments which is why you are seeing these failures

Reply:
Should I remove it because of the precision error? or I can do something like this by adding a comment:
```
#This test may fail locally because of different precision in some environment
assert str(csl) == '1:3.40282346639e+38' # Actual value is '1:3.4028234663852886e+38'
```
This will pass the CI tests

Reply:
So, I just had a closer look at the `cbandit_label` class, and it is actually incorrectly implemented. Thanks for finding this! I've created an issue here: #2295. Getting the string function to output the correct fields is critical to making sure the test is right. Would you like to work on that issue and submit a different PR for it?

Also, with this test, the label that is being parsed is missing the other two components of a CB label. Generally they come in the form: `action:cost:probability`. It seems the test was originally written just looking at the action, but since you are testing more fields you should expand the label to all 3. See [here](https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Logged-Contextual-Bandit-Example) for example of these CB labels.

Also, just a side note tests must pass both locally and in CI.

Reply:
> Would you like to work on that issue and submit a different PR for it?

Yes!, I will take it and submit a PR.